components/TreeHeatmap/index.js

```diff
@@ -119,16 +124,20 @@ export default class TreeHeatmap extends VisComponent {
 
     let keys = Object.keys(this.data[0]);
     let idColumn = this.idColumn;
-    if (!idColumn) {
+    if (idColumn === undefined) {
       idColumn = keys.includes('_id') ? '_id' : idColumn;
       idColumn = keys.includes('_') ? '_' : idColumn;
       idColumn = keys.includes('') ? '' : idColumn;
     }
+    if (idColumn === undefined) {
+      console.log('TreeHeatmap: No suitable idColumn found.');
+      return;
+    }
 
     let rows = [];
     let reachedMetadata = false;
     this.data.forEach(row => {
-      let id = row[idColumn];
+      let id = '' + row[idColumn];
```

Comment:
That's so weird - do you know why `String(x)` can't just do whatever `'' + x` does under the hood?

In any case, @jeffbaumes, maybe we should revert this particular change in light of @manthey's performance concerns.